testng-core/src/test/java/test/configuration/issue2432/IssueTest.java

```diff
@@ -34,9 +34,9 @@ public void ensureNoImplicitDependencyIsAddedWhenGroupsAreInvolved() {
     List<String> expected =
         Arrays.asList(
             "prepareConfig",
+            "prepareConfigForTest1",
```

Comment:
Is that change expected?

Reply:
yes, the test is totally wrong and change how DependsOnGroup works. 
#2432 change so there is no longer possible to have dependencies between test class and base class, which was possible before.

Reply:
In my opinion I think that this class hierarchy is kind of strange behavior, which would have been better to solve with depends on and Priority.